### PR TITLE
Feature/wysiwyg

### PR DESCRIPTION
**Pull request by Vercjames**
WYSIWYG Validation Fixed - FOR-975
WYSIWYG Spell Check Added -  FOR-929

### CODE REVIEW

File: src/components/textarea.js [hunk 1]
```diff
@@ -1,5 +1,18 @@
 var fs = require('fs');
 module.exports = function(app) {
+  app.directive('ckeditor', function() {
```
Comment (travist) [inline src/components/textarea.js]: this causes a name collision with the other ```ckeditor``` directive that we use. I would prefer to call this ```formio-ckeditor```

Comment (travist) [reply src/components/textarea.js]: This would make then name of this become ```formioCkeditor```, which would then be implemented as ```formio-ckeditor``` in the files.

File: src/templates/components/textarea.html [hunk 2]
```diff
@@ -3,6 +3,7 @@
   ng-model="data[component.key]"
   ng-disabled="readOnly"
   ng-required="isRequired(component)"
+  spellcheck="{{ component.spellcheck }}"
```
Comment (travist) [inline src/templates/components/textarea.html]: We need to separate out this into a different pull request so that it does not get hung up in the review process.

File: bower.json [hunk 3]
```diff
@@ -26,7 +26,8 @@
     "angular-ui-ace": "^0.2.3",
     "angular-ui-select": "^0.19.8",
     "angular-bootstrap": "^2.5.0",
-    "angular-ckeditor": "^1.0.3"
+    "angular-ckeditor": "^1.0.3",
+    "ckeditor": "^4.8.0"
```
Comment (travist) [inline bower.json]: This will have negative implications on our renderer size since people will now need to include ALL ckeditor files regardless if they need it or now. Previously, we have told people that if you want ckeditor, then just include it from CDN within your application. Not sure why we cannot keep doing that.

File: package.json [hunk 4]
```diff
@@ -44,7 +44,6 @@
   },
   "dependencies": {
     "angular": "^1.6.6",
-    "angular-ckeditor": "lemonde/angular-ckeditor#v1.0.3",
```
Comment (travist) [inline package.json]: Why was this removed?

Comment (Vercjames) [reply package.json]: Isn't this redundant. We pull angular-ckeditor from the bower.js file.

Comment (travist) [reply package.json]: Bower and NPM are two different ways to pull in this module. It's redundant because we provide people a choice which one they want to use.
